packages/sdk-react/src/components/auth/OtpVerification.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { useRef, useState } from "react";
+import OtpInput from "./otp";
+import styles from "./OtpVerification.module.css";
+import { formatPhoneNumber } from "./utils";
+import { otpAuth } from "../../actions";
+import EmailIcon from "@mui/icons-material/Email";
+import SmsIcon from "@mui/icons-material/Sms";
+import { CircularProgress } from "@mui/material";
+interface OtpVerificationProps {
+  type: string;
+  contact: string;
+  suborgId: string;
+  otpId: string;
+  authIframeClient: any;
+  onValidateSuccess: (credentialBundle: any) => Promise<void>;
+  onResendCode: (
+    type: "EMAIL" | "PHONE_NUMBER",
+    value: string
+  ) => Promise<void>;
+}
+
+const OtpVerification: React.FC<OtpVerificationProps> = ({
+  type,
+  contact,
+  suborgId,
+  otpId,
+  authIframeClient,
+  onValidateSuccess,
+  onResendCode,
+}) => {
+  const [otpError, setOtpError] = useState<string | null>(null);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [resendText, setResendText] = useState("Resend code");
+  const otpInputRef = useRef<any>(null);
+
+  const handleValidateOtp = async (otp: string) => {
+    setOtpError(null);
+    setIsLoading(true);
+    try {
+      const authResponse = await otpAuth({
+        suborgID: suborgId,
+        otpId,
+        otpCode: otp,
+        targetPublicKey: authIframeClient!.iframePublicKey!,
+      });
+
+      if (authResponse?.credentialBundle) {
+        await onValidateSuccess(authResponse.credentialBundle);
+      } else {
+        setOtpError("Invalid code. Please try again.");
+      }
+      otpInputRef.current.resetOtp();
+    } catch (error) {
+      setOtpError("An error occurred. Please try again.");
+    }
+    setIsLoading(false);
+  };
+
+  const handleResendCode = async () => {
+    setOtpError(null);
+    try {
+      await onResendCode(
+        type === "otpEmail" ? "EMAIL" : "PHONE_NUMBER",
+        contact
+      );
+      setResendText("Code sent ✓");
+
+      setTimeout(() => {
+        setResendText("Resend code");
+      }, 15000);
```

Comment:
might want to bring more attention to this default value (e.g. by putting it into a const/defining it at the top)